### PR TITLE
Fix a bug that the output of `tf.sparse.fill_empty_rows` may not be ordered

### PR DESCRIPTION
**Pull request by firejq**
This is a PR from JIZHI Team & TaiJi AI platform in Tencent.

This pull request is for tf 2.2.0, to fix a bug that the `sp_ordered_output ` of `tf.sparse.fill_empty_rows` is not ordered when the sp_input is unordered and all rows of the sp_input are not empty.

Referring to the [docs](https://www.tensorflow.org/api_docs/python/tf/sparse/fill_empty_rows), the output of `tf.sparse.fill_empty_rows` named `sp_ordered_output` is ordered. e.g.
```python
sp_ids = tf.SparseTensor(indices=[[1, 2], [1, 3], [0, 1], [0, 3]],
                         values=[2, 1, 1, 1],
                         dense_shape=[3, 4])
sp_ids, _ = tf.sparse.fill_empty_rows(sp_ids, default_value=0)
print(sp_ids)
```
Output:
```
SparseTensor(
  indices=tf.Tensor(
  [[0 1]
   [0 3]
   [1 2]
   [1 3]
   [2 0]], shape=(5, 2), dtype=int64), 
  values=tf.Tensor([1 1 2 1 0], shape=(5,), dtype=int32), 
  dense_shape=tf.Tensor([3 4], shape=(2,), dtype=int64)
)
```

However, the `sp_ordered_output` is not ordered when the sp_input is unordered and all rows of the sp_input are not empty. e.g.
```python
sp_ids = tf.SparseTensor(indices=[[1, 2], [1, 3], [0, 1], [0, 3]],
                         values=[2, 1, 1, 1],
                         dense_shape=[2, 4])
sp_ids, _ = tf.sparse.fill_empty_rows(sp_ids, default_value=0)
print(sp_ids)
```
Output:
```
SparseTensor(
  indices=tf.Tensor(
  [[1 2]
   [1 3]
   [0 1]
   [0 3]], shape=(4, 2), dtype=int64), 
  values=tf.Tensor([2 1 1 1], shape=(4,), dtype=int32), 
  dense_shape=tf.Tensor([2 4], shape=(2,), dtype=int64)
)
```

So we add a judgment for whether the `sp_input` is ordered, and turns to make a new ordered output tensor instead of returning the original one.

Thank you for your time on reviewing this PR.

### CODE REVIEW

**Comment by firejq** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here with `@googlebot I signed it!` and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Ftensorflow%2Ftensorflow%2Fpull%2F39678) for more info**.

@googlebot I signed it!

**Comment by gbaned** [comment]
@firejq Thank you for your contribution. Can you please sign CLA? Thanks!

**Comment by firejq** [comment]
@gbaned I have signed it [here](https://cla.developers.google.com/), could you help me to verify it? Thanks!

**Comment by firejq** [comment]
@gbaned The [CLA page](https://cla.developers.google.com/) has showed I signed it. If I have any incorrect operation, can you please point out? Thanks!

**Comment by firejq** [comment]
@googlebot I signed it!

**Comment by firejq** [comment]
@gbaned Could you have a look please?

**Comment by firejq** [comment]
@ebrevdo Could you have a look please?
Thanks!

**Comment by gbaned** [comment]
@ebrevdo Can you please review this PR ? Thanks!

File: tensorflow/core/kernels/sparse_fill_empty_rows_op.cc [hunk 1]
```diff
@@ -118,13 +118,17 @@ class SparseFillEmptyRowsOp : public OpKernel {
       return;
     }
 
+    bool indices_is_order = true;
```
Comment (ebrevdo) [inline tensorflow/core/kernels/sparse_fill_empty_rows_op.cc:121]: rename to `rows_are_ordered`

Comment (firejq) [reply tensorflow/core/kernels/sparse_fill_empty_rows_op.cc:121]: Ok!

File: tensorflow/core/kernels/sparse_fill_empty_rows_op.cc [hunk 2]
```diff
@@ -147,7 +151,7 @@ class SparseFillEmptyRowsOp : public OpKernel {
       }
     }
 
-    if (all_rows_full) {
+    if (all_rows_full && rows_are_ordered) {
```
Comment (ebrevdo) [inline tensorflow/core/kernels/sparse_fill_empty_rows_op.cc:154]: this needs a unit test - there are unit tests in python/kernel_tests/

Comment (firejq) [reply tensorflow/core/kernels/sparse_fill_empty_rows_op.cc:154]: Hi, I have added a unit test `testNoEmptyRowsAndUnordered` in python/kernel_tests/sparse_ops_test.py.  Could you please review again? Thanks!

**Review by ebrevdo — APPROVED** [review]

**Comment by firejq** [comment]
Hi, @gbaned, @ebrevdo . 
Could you please have a look at the reason for `import/copybara` checking fail?  I would like to fix if there is something wrong in my PR.

**Comment by gbaned** [comment]
> Hi, @gbaned, @ebrevdo .
> Could you please have a look at the reason for `import/copybara` checking fail? I would like to fix if there is something wrong in my PR.

@firejq  It is successfully completed now. Thank you.

**Comment by firejq** [comment]
@gbaned Could you help to merge this PR? Thanks!

**Comment by gbaned** [comment]
> @gbaned Could you help to merge this PR? Thanks!

@firejq This is only waiting for an internal approval, it will be merged soon.

**Pull request merged**